prov/gni/src/gnix_rma.c

```diff
@@ -829,24 +837,124 @@ int _gnix_rma_post_rdma_chain_req(void *data)
 	return FI_SUCCESS;
 }
 
+/* Fill head and tail descriptors for chained gets */
+static void __gnix_rma_more_fill_sub_htd(struct gnix_fab_req *req,
+					gni_ct_get_post_descriptor_t *more_get,
+					gni_mem_handle_t *rem_mdh,
+					int *idx, int *entries)
+{
+	int head_off, tail_len;
+	struct gnix_fid_ep *ep = req->gnix_ep;
+
+	head_off = req->rma.rem_addr & GNI_READ_ALIGN_MASK;
+	tail_len = (req->rma.rem_addr + req->rma.len)
+		& GNI_READ_ALIGN_MASK;
+
+	/* Get int_tx_buf if unaligned */
+	if ((req->int_tx_buf_e == NULL) && (head_off || tail_len)) {
+		req->int_tx_buf_e = _gnix_ep_get_int_tx_buf(ep);
+		if (req->int_tx_buf == NULL) {
+			GNIX_FATAL(FI_LOG_EP_DATA, "RAN OUT OF INT_TX_BUFS");
+			/* TODO Create growable list of buffers */
+		}
+	} else {
+		return;
+	}
+
+	req->int_tx_buf = ((struct gnix_int_tx_buf *)
+				req->int_tx_buf_e)->buf;
+	req->int_tx_mdh = _gnix_ep_get_int_tx_mdh(ep);
+
+	if (head_off) {
+		more_get[*idx].ep_hndl = req->vc->gni_ep;
+		more_get[*idx].length = GNI_READ_ALIGN;
+		more_get[*idx].remote_addr =
+				req->rma.rem_addr & ~GNI_READ_ALIGN_MASK;
+		more_get[*idx].remote_mem_hndl = *rem_mdh;
+		more_get[*idx].local_addr = (uint64_t)req->int_tx_buf;
+		more_get[*idx].local_mem_hndl = req->int_tx_mdh;
+
+		if (*idx < (*entries) - 1)
+			more_get[*idx].next_descr = &more_get[(*idx) + 1];
+		else
+			more_get[*idx].next_descr = NULL;
```

Comment:
Is it possible to have used all the entries and have `tail_len != 0`?  If so, isn't there a problem in this code?